src/main/kotlin/controllers/CacheChooserController.kt

```diff
@@ -68,32 +69,34 @@ class CacheChooserController(
             isVisible = false
             selectionMode = ListSelectionModel.SINGLE_SELECTION
             addListSelectionListener {
-                if (selectedIndex != -1) {
-                    btnDownload.isEnabled = true
-                }
+                btnDownload.isEnabled = selectedIndex != -1
             }
         }
         val txtCacheLocation = JTextField().apply {
             isEnabled = false
             maximumSize = Dimension(maximumSize.width, preferredSize.height)
         }
         val lblStatusText = JLabel()
+        val lblErrorText = JLabel().apply {
+            foreground = Color.RED
+        }
         btnDownload.addActionListener {
             btnDownload.isEnabled = false
             listCaches.selectedValue?.let {
                 lblStatusText.text = "Downloading cache $it, please wait.."
                 txtCacheLocation.text = ""
 
-                downloadCache(it) { path ->
+                downloadCache(it, { path ->
                     lblStatusText.text = ""
                     txtCacheLocation.text = path
                     btnDownload.isEnabled = true
-                }
+                }, { err ->
+                    lblStatusText.text = ""
+                    lblErrorText.text = "Failed to download cache: $err"
+                    btnDownload.isEnabled = true
+                })
```

Comment:
Does this address my comment on this issue:
https://github.com/ConnorDY/OSRS-Environment-Exporter/issues/57#issuecomment-1212690497